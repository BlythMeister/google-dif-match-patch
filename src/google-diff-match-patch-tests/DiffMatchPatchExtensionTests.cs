using ApprovalTests;
using ApprovalTests.Reporters;
using DiffMatchPatch;
using NUnit.Framework;

namespace DiffMatchPatchTests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class DiffMatchPatchExtensionTests
    {
        private const string expected = "start\n" +
                                        "line with some content\n" +
                                        "unchanged content\n" +
                                        "removed line\n" +
                                        "end";

        private const string actual = "start\n" +
                                      "new line\n" +
                                      "line with some modified content\n" +
                                      "unchanged content\n" +
                                      "end";

        private const string expectedHtml = "<p>start</p>\n" +
                                            "<p>line with some content</p>\n" +
                                            "<p>unchanged content</p>\n" +
                                            "<p>removed line</p>\n" +
                                            "<p>end</p>";

        private const string actualHtml = "<p>start</p>\n" +
                                          "<p>new line</p>\n" +
                                          "<p>line with some modified content</p>\n" +
                                          "<p>unchanged content</p>\n" +
                                          "<p>end</p>";

        [Test]
        public void CorrectPrettyTextOutput_TextInput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expected, actual);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var (patchCount, prettyOutput) = dmp.diff_toPrettyText(diffs);

            Approvals.Verify(prettyOutput);
            Assert.AreEqual(3, patchCount);
        }

        [Test]
        public void CorrectPrettyHtmlOutput_TextInput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expected, actual);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var prettyOutput = dmp.diff_toPrettyHtml(diffs);

            Approvals.Verify(prettyOutput);
        }

        [Test]
        public void CorrectPrettyHtmlDocOutput_TextInput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expected, actual);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var prettyOutput = dmp.diff_toPrettyHtmlDoc(diffs);

            Approvals.Verify(prettyOutput);
        }

        [Test]
        public void CorrectPrettyTextOutput_HtmlInput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expectedHtml, actualHtml);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var (patchCount, prettyOutput) = dmp.diff_toPrettyText(diffs);

            Approvals.Verify(prettyOutput);
            Assert.AreEqual(3, patchCount);
        }

        [Test]
        public void CorrectPrettyHtmlOutput_HtmlInput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expectedHtml, actualHtml);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var prettyOutput = dmp.diff_toPrettyHtml(diffs);

            Approvals.Verify(prettyOutput);
        }

        [Test]
        public void CorrectPrettyHtmlDocOutput_HtmlInput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expectedHtml, actualHtml);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var prettyOutput = dmp.diff_toPrettyHtmlDoc(diffs);

            Approvals.Verify(prettyOutput);
        }
    }
}
