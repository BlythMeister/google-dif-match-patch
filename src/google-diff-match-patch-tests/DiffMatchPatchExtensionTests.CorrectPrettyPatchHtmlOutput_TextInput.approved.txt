<html>
<head>
</head>
<body style="font-family: 'Lucida Console', Courier, monospace;">

<!-- START OF PATCHES -->

<div style="border: 1px solid black; margin-bottom: 5px; padding: 5px;">
<div style="padding: 5px 0;">
<div>Patch Number: 1</div>
<div>Delete Character Coordinates: N/A</div>
<div>Insert Character Coordinates: 1 - 28</div>
</div>
<div style="border-top: 1px solid black; padding: 5px 0;">
<span>
start<br />

</span>
<ins style="background:#e6ffe6;">
new line<br />

</ins>
<span>
line with so
</span>
</div>
</div>
<div style="border: 1px solid black; margin-bottom: 5px; padding: 5px;">
<div style="padding: 5px 0;">
<div>Patch Number: 2</div>
<div>Delete Character Coordinates: N/A</div>
<div>Insert Character Coordinates: 19 - 52</div>
</div>
<div style="border-top: 1px solid black; padding: 5px 0;">
<span>
e with some 
</span>
<ins style="background:#e6ffe6;">
modified 
</ins>
<span>
content<br />
unch
</span>
</div>
</div>
<div style="border: 1px solid black; margin-bottom: 5px; padding: 5px;">
<div style="padding: 5px 0;">
<div>Patch Number: 3</div>
<div>Delete Character Coordinates: 60 - 82</div>
<div>Insert Character Coordinates: N/A</div>
</div>
<div style="border-top: 1px solid black; padding: 5px 0;">
<span>
ntent<br />

</span>
<del style="background:#ffe6e6;">
removed line<br />

</del>
<span>
end
</span>
</div>
</div>
<!-- END OF PATCHES -->

</body>
</html>